src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -88,102 +92,148 @@ static bool GetCommandAndKey(
             StackExchangeRedisConnectionInstrumentation.CreationTags,
             startTime: command.CommandCreated);
 
-        if (activity == null)
+        if (activity is null && metrics.Enabled is false)
         {
             return null;
         }
 
-        activity.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        activity?.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        var meterTags = metrics.Enabled ?
+            new TagList(StackExchangeRedisConnectionInstrumentation.CreationTags) :
+            default(IList<KeyValuePair<string, object?>>);
 
-        if (activity.IsAllDataRequested)
-        {
-            // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+        // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+
+        // Timing example:
+        // command.CommandCreated; //2019-01-10 22:18:28Z
 
-            // Timing example:
-            // command.CommandCreated; //2019-01-10 22:18:28Z
+        // command.CreationToEnqueued;      // 00:00:32.4571995
+        // command.EnqueuedToSending;       // 00:00:00.0352838
+        // command.SentToResponse;          // 00:00:00.0060586
+        // command.ResponseToCompletion;    // 00:00:00.0002601
 
-            // command.CreationToEnqueued;      // 00:00:32.4571995
-            // command.EnqueuedToSending;       // 00:00:00.0352838
-            // command.SentToResponse;          // 00:00:00.0060586
-            // command.ResponseToCompletion;    // 00:00:00.0002601
+        // Total:
+        // command.ElapsedTime;             // 00:00:32.4988020
 
-            // Total:
-            // command.ElapsedTime;             // 00:00:32.4988020
+        var flags = command.Flags.ToString();
+        activity?.SetTag(SemanticConventions.AttributeDbRedisFlagsKeyName, flags);
+        meterTags?.Add(SemanticConventions.AttributeDbRedisFlagsKeyName, flags);
```

Comment:
this is not part of metric tags, so should not be added.

Reply:
removed

Reply:
@cijothomas could you resume the review to see if it meets expectations?

Reply:
sure will do today.